web-common/src/features/canvas/stores/time-control.ts

```diff
@@ -213,106 +215,38 @@ export class TimeControls {
       },
     );
 
-    this.timeRangeText = derived(
-      [
-        this.selectedTimeRange,
-        this.selectedComparisonTimeRange,
-        this.showTimeComparison,
-      ],
-      ([
-        selectedTimeRange,
-        selectedComparisonTimeRange,
-        showTimeComparison,
-      ]) => {
-        const searchParams = new URLSearchParams();
-
-        searchParams.set(
-          ExploreStateURLParams.TimeRange,
-          toTimeRangeParam(selectedTimeRange),
-        );
-
-        if (showTimeComparison && selectedComparisonTimeRange) {
-          searchParams.set(
-            ExploreStateURLParams.ComparisonTimeRange,
-            toTimeRangeParam(selectedComparisonTimeRange),
-          );
-        }
-
-        if (selectedTimeRange?.interval) {
-          const mappedTimeGrain =
-            ToURLParamTimeGrainMapMap[selectedTimeRange?.interval];
-          if (mappedTimeGrain) {
-            searchParams.set(ExploreStateURLParams.TimeGrain, mappedTimeGrain);
-          }
-        }
-
-        return searchParams.toString();
-      },
-    );
-
-    this.setInitialState();
-  }
-
-  setInitialState = () => {
-    const defaultStore = derived(
-      [this.allTimeRange, this.specStore],
-      ([allTimeRange, spec]) => {
-        if (!spec?.data || allTimeRange.isFetching || this.isInitialStateSet) {
-          return;
-        }
-
-        const isLocalComponentControl = Boolean(this.componentName);
-
-        const selectedTimezone = get(this.selectedTimezone);
-        const comparisonTimeRange = get(this.selectedComparisonTimeRange);
+    this.searchParamsStore.subscribe((searchParams) => {
+      this.setTimeFiltersFromText(searchParams.toString());
+    });
 
-        const { defaultPreset } = spec.data?.canvas || {};
+    if (!componentName) {
+      this.specStore.subscribe((spec) => {
+        const defaultPreset = spec.data?.canvas?.defaultPreset;
         const timeRanges = spec?.data?.canvas?.timeRanges;
+        if (!defaultPreset) return;
```

Comment:
If I do not have a default time range set, the global header crashes 
<img width="697" height="141" alt="image" src="https://github.com/user-attachments/assets/1b9b1d5f-e38e-49c5-816f-46326002e4b7" />

On explore, setting `default_time_range` is optional as well.